tests/loader_regression_tests.cpp

```diff
@@ -448,6 +448,404 @@ TEST_F(EnumeratePhysicalDevices, TwoCallIncomplete) {
+    ASSERT_EQ(VK_SUCCESS, inst->vkEnumeratePhysicalDevices(inst, &returned_physical_count, physical_device_handles_1.data()));
+    ASSERT_EQ(physical_count, returned_physical_count);
+
+    // Delete the 2nd physical device
+    driver.physical_devices.erase(std::next(driver.physical_devices.begin()));
+
+    physical_count = static_cast<uint32_t>(driver.physical_devices.size());
+    std::vector<VkPhysicalDevice> physical_device_handles_2 = std::vector<VkPhysicalDevice>(returned_physical_count);
+    ASSERT_EQ(VK_SUCCESS, inst->vkEnumeratePhysicalDevices(inst, &returned_physical_count, physical_device_handles_2.data()));
+    ASSERT_EQ(physical_count, returned_physical_count);
+    physical_device_handles_2.resize(returned_physical_count);
+
+    returned_physical_count = physical_count;
+    std::vector<VkPhysicalDevice> physical_device_handles_3 = std::vector<VkPhysicalDevice>(returned_physical_count);
+    ASSERT_EQ(VK_SUCCESS, inst->vkEnumeratePhysicalDevices(inst, &returned_physical_count, physical_device_handles_3.data()));
+    ASSERT_EQ(physical_count, returned_physical_count);
+
+    // Make sure one has 1 more device that two or three
+    ASSERT_EQ(physical_device_handles_1.size(), physical_device_handles_2.size() + 1);
+    ASSERT_EQ(physical_device_handles_1.size(), physical_device_handles_3.size() + 1);
+
+    // Make sure the devices in two and three are all found in one
+    uint32_t two_found = 0;
+    uint32_t three_found = 0;
+    for (uint32_t count = 0; count < physical_device_handles_1.size(); ++count) {
+        for (uint32_t int_count = 0; int_count < physical_device_handles_2.size(); ++int_count) {
+            if (physical_device_handles_2[int_count] == physical_device_handles_1[count]) {
+                two_found++;
+                break;
+            }
+        }
+        for (uint32_t int_count = 0; int_count < physical_device_handles_3.size(); ++int_count) {
+            if (physical_device_handles_3[int_count] == physical_device_handles_1[count]) {
+                three_found++;
+                break;
+            }
+        }
+    }
+    ASSERT_EQ(two_found, returned_physical_count);
+    ASSERT_EQ(three_found, returned_physical_count);
+}
+
+TEST_F(EnumeratePhysicalDevices, MultipleAddRemoves) {
+    auto& driver = env->get_test_icd().SetMinICDInterfaceVersion(5);
+
+    driver.physical_devices.emplace_back("physical_device_0");
+    driver.physical_devices.emplace_back("physical_device_1");
+    driver.physical_devices.emplace_back("physical_device_2");
+    driver.physical_devices.emplace_back("physical_device_3");
+
+    InstWrapper inst{env->vulkan_functions};
+    inst.CheckCreate();
+
+    uint32_t physical_count = static_cast<uint32_t>(driver.physical_devices.size());
+    uint32_t returned_physical_count = physical_count;
+    std::vector<VkPhysicalDevice> physical_device_handles_1 = std::vector<VkPhysicalDevice>(physical_count);
```

Comment:
Nit/style comment.
Would be nice to put `physical_device_handles_1` into an array of 8 like `std::array<std::vector<VkPhysicalDevice>, 8>` so you aren't defining variables all over the place.
Then the final 'assert' could be a loop over it with the values you assert are true.

alternatively, create a new struct which holds the vector and the value it should have at the end, so you aren't having to count them up at the end.